profiler/profile.go

```diff
@@ -216,6 +229,108 @@ func goroutineProfile(cfg *config) (*profile, error) {
 	}, nil
 }
 
+func goroutineWaitProfile(cfg *config) (*profile, error) {
+	var (
+		text  = &bytes.Buffer{}
+		pprof = &bytes.Buffer{}
+		start = now()
+	)
+	if err := lookupProfile(GoroutineProfile.String(), text, 2); err != nil {
+		return nil, err
+	} else if err := goroutineDebug2ToPprof(text, pprof); err != nil {
+		return nil, err
+	}
+	end := now()
+	tags := append(cfg.tags, GoroutineWaitProfile.Tag())
+	cfg.statsd.Timing("datadog.profiler.go.collect_time", end.Sub(start), tags, 1)
+
+	return &profile{
+		name: GoroutineWaitProfile.Filename(),
+		data: pprof.Bytes(),
+	}, nil
+}
+
+func goroutineDebug2ToPprof(r io.Reader, w io.Writer) error {
+	goroutines, errs := stackparse.Parse(r)
+
+	functionID := uint64(1)
+	locationID := uint64(1)
+
+	p := &pprofile.Profile{}
+	m := &pprofile.Mapping{ID: 1, HasFunctions: true}
+	p.Mapping = []*pprofile.Mapping{m}
+	p.SampleType = []*pprofile.ValueType{
+		{
+			Type: "waitduration",
+			Unit: "nanoseconds",
+		},
+	}
+
+	for _, g := range goroutines {
+		// TODO(fg) exclude goroutines w/ g.Wait == 0?
+
+		sample := &pprofile.Sample{
+			Value: []int64{g.Wait.Nanoseconds()},
+			Label: map[string][]string{
+				"state": {g.State}, // TODO(fg) split into atomicstatus/waitreason?
+			},
+			NumUnit:  map[string][]string{"goid": {"id"}},
+			NumLabel: map[string][]int64{"goid": {int64(g.ID)}},
+			// TODO(fg) add g.LockedToThread, g.CreatedBy, ...?
```

Comment:
@pmbauer what do you think?

I think adding CreatedBy like it was part of the `g.Stack` might be useful for people? OTOH the existing GoroutineProfile doesn't do this, so maybe it's more confusing than useful?

For LockedToThread: Do you see any issue with just throwing that into a label as well?

Reply:
I'd err on the side of adding more context than less

Reply:
208ef48 - Done